pandas/tests/groupby/conftest.py

```diff
@@ -76,3 +76,49 @@ def three_group():
                       'D': np.random.randn(11),
                       'E': np.random.randn(11),
                       'F': np.random.randn(11)})
+
+
+@pytest.fixture
+def df_cat():
+    df = DataFrame({'a': ['x', 'x', 'x', 'y'],
+                    'b': ['a', 'a', 'b', 'a'],
+                    'c': [1, 2, 3, 4]})
+    df['a'] = df['a'].astype('category')
+    df['b'] = df['b'].astype('category')
+    return df
+
+
+@pytest.fixture
+def multi_index_cat_complete():
+    lvls = [CategoricalIndex(['x', 'y'], categories=['x', 'y'], ordered=False),
+            CategoricalIndex(['a', 'b'], categories=['a', 'b'], ordered=False)]
+    index = MultiIndex.from_product(lvls, names=['a', 'b'])
+    return index
+
+
+@pytest.fixture
+def multi_index_cat_partial(df_cat):
```

Comment:
Question on naming - what does `partial` mean here?

Reply:
By `partial` I mean that it's not a full product of index level values. For example, if first level has `['x', 'y']` and second level `['a', 'b']`, then a complete index is `[('x', 'a'), ('x', 'b'), ('y', 'a'), ('y', 'b')]` and I construct that with `.from_product`, and if not all combinations are in, then it's `partial` (for example, `[('x', 'a'), ('x', 'b'), ('y', 'a')]` where `('y', 'b')` is not in.

Reply:
Hmm OK. I find this rather confusing though that the determination as to whether or not this is "partial" is done entirely outside the scope of the fixture (i.e. it is up to the injected test to only partially align).

Is there not a way to make the fixture self contained to provide a Series or Frame instead? May require a rewrite of your tests but I think this is just confusing